src/interface/viewport.c

```diff
@@ -2636,8 +2636,216 @@ void store_interaction_info(paint_struct *ps)
  *
  *  rct2: 0x00679074
  */
-void sub_679074(rct_drawpixelinfo *dpi, int imageId, int x, int y)
+void sub_679074(rct_drawpixelinfo *dpi, int imageId, sint16 x, sint16 y)
 {
+	rct_g1_element image = g1Elements[imageId & 0x7FFFF];
+
+	switch (dpi->zoom_level) {
+		case 0:
+			// TODO: loc_6790A0:
+			break;
+
+		case 1:
+			if (image.flags & 0x20) {
+				return;
+			}
+
+			if (!(image.flags & 0x10)) {
+				dpi->zoom_level--;
+				dpi->x /= 2;
+				dpi->y /= 2;
+
+				sub_679074(dpi, imageId - image.zoomed_offset, x / 2, y / 2);
+
+				dpi->zoom_level++;
+				dpi->x *= 2;
+				dpi->y *= 2;
+
+				return;
+			}
+
+			RCT2_GLOBAL(0x9E3D08, uint32) = image.offset;
+			RCT2_GLOBAL(0x9E3D0C, sint16) = image.width;
+			RCT2_GLOBAL(0x9E3D0C + 2, sint16) = image.height;
+			RCT2_GLOBAL(0x9E3D10, sint16) = image.x_offset;
+			RCT2_GLOBAL(0x9E3D10 + 2, sint16) = image.y_offset;
+			RCT2_GLOBAL(0x9E3D14, uint16) = image.flags;
+			RCT2_GLOBAL(0x9E3D14 + 2, uint16) = image.zoomed_offset;
+
+			if (image.flags & 4) {
+				// push edi
+				y--;
+				// ebp = edi;
+				// esi = image.offset;
+				y += image.y_offset;
+
+				uint16 height = image.height;
+				RCT2_GLOBAL(RCT2_ADDRESS_Y_START_POINT_GLOBAL, uint32) = 0;
+				if (height % 2) {
+					height--;
+					if (height == 0) {
+						return;
+					}
+
+					RCT2_GLOBAL(RCT2_ADDRESS_Y_START_POINT_GLOBAL, uint32)++;
+				}
+
+
+				// TODO: (y = y/2?)
+				y &= 0xFFFE;
```

Comment:
Suggest just using https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/drawing/sprite.c#L491 as there is no point duplicating this code.